polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -480,6 +487,7 @@ impl<T: Config> Pallet<T> {
 			}
 
 			ensure!(all_weight_before.all_lte(max_block_weight), Error::<T>::InherentOverweight);
+			ensure!(!statements_were_dropped, Error::<T>::InherentOverweight);
```

Comment:
do we want to reject blocks where at least one statement is dropped? what do we do with disputes?

Reply:
Afair the else statement is executed during block execution only (right?) and this is a sanity check that we are not importing votes from disabled validators. We discussed this [here](https://github.com/paritytech/polkadot-sdk/issues/1592#issuecomment-1722234182).